algo-builder-tester/test/algob-tester.ts

```diff
@@ -151,74 +153,141 @@ describe.skip('algob-tester', function () {
-      sign: types.SignType.SecretKey,
-      fromAccount: admin.account,
-      payFlags: { totalFee: fee },
-      toAccountAddr: admin.address,
-      amountMicroAlgos: BigInt(1e3),
-    }
-    assert.throws(() => { runtime.executeTx([callAppParams, escrowTxParams]) }, 'RUNTIME_ERR1007: Teal code rejected by logic')
-    // runtime.executeTx();
-  })
-})
\ No newline at end of file
+    callAppParams.appArgs = ["str:TST8", "str:usd"];
+    runtime.executeTx(callAppParams);
+    syncAccounts();
+    assert.equal(fetchGlobalBytes("console"), "coin_name not found");
+
+    callAppParams.appArgs = ["str:TST8", "str:atom"];
+    runtime.executeTx(callAppParams);
+    syncAccounts();
+    assert.equal(fetchGlobalBytes("console"), "atom");
+  });
+  describe.only("capability test, 2022Feb27 batch, appArgs as name, +TST7,8", function () {
+    it("LOG: test log message", function () {
+      const callAppParams: types.AppCallsParam = {
+        type: types.TransactionType.CallApp,
+        sign: types.SignType.SecretKey,
+        fromAccount: alice.account,
+        appID: appID,
+        payFlags: { totalFee: fee },
+        appArgs: ["str:LOG"],
+      };
+      // let app = runtime.getApp(appID); // not in global state nor app.
+      // syncAccounts(); let als = alice.getAppFromLocal(appID); // not in local state
+      let rct = runtime.executeTx(callAppParams); // not in receipt
+      if (Array.isArray(rct)) rct = rct[0]; // for TS Engine
+      assert.deepEqual(rct.logs!, ["I'm a sample log"]);
+    });
+    it("EXT_GLOBAL: test log message", function () {
+      const verifierID = runtime.deployApp(
+        "price-verifier.py",
+        runtime_config.clearProgramFileName,
+        {
+          sender: admin.account,
+          globalBytes: 1,
+          globalInts: 1,
+          localBytes: 0,
+          localInts: 0,
+        },
+        {}
+      ).appID;
+
+      const callAppParams: types.AppCallsParam = {
+        type: types.TransactionType.CallApp,
+        sign: types.SignType.SecretKey,
+        fromAccount: alice.account,
+        appID: appID,
+        payFlags: { totalFee: fee },
+        appArgs: ["str:EXT_GLOBAL"],
+        foreignApps: [verifierID],
+      };
+      runtime.executeTx(callAppParams);
+      syncAccounts();
+      assert.equal(admin.getGlobalState(appID, "var1"), 3456789n);
+      assert.equal(fetchGlobalBytes("console"), "1234567890-1234");
+
+      // changing price
+      const callVerifierParams: types.AppCallsParam = {
+        type: types.TransactionType.CallApp,
+        sign: types.SignType.SecretKey,
+        fromAccount: admin.account,
+        appID: verifierID,
+        payFlags: { totalFee: fee },
+        appArgs: ["str:irrelevant", "str:asd89yhg890"],
+        // TODO:discuss#1: how to pass Int?
+        // appArgs: ["int:10348967", "str:asd89yhg890"],
+        // appArgs: ["str:10348967n", "int:10348967"],
+        // appArgs: [new Uint8Array(10348967), "int:10348967"],
```

Comment:
I think: use int:10348967. It will be interpreted as a bytes, then use this https://developer.algorand.org/docs/get-details/dapps/avm/teal/opcodes/#extract_uint64

to get a bytes as uint64